telegram/inputfile.py

```diff
@@ -67,7 +67,10 @@ def __init__(self,
 
         if isinstance(self.input_file, file):
             self.input_file_content = self.input_file.read()
-            self.filename = os.path.basename(self.input_file.name)
+            if 'filename' in self.data and self.data['filename']:
```

Comment:
``and self.data['filename']`` is redundant of ``if 'filename' in self.data``, so you can keep only the first part.